magnum-zio-example/src/main/scala/com/augustnagro/magnum/magzio/example/ExampleZioApp.scala

```diff
@@ -0,0 +1,80 @@
+package com.augustnagro.magnum.magzio.example
+
+import com.augustnagro.magnum.magzio.*
+import com.dimafeng.testcontainers.PostgreSQLContainer
+import org.postgresql.ds.PGSimpleDataSource
+import org.testcontainers.utility.DockerImageName
+import zio.*
+
+import javax.sql.DataSource
+
+object ExampleZioApp extends ZIOAppDefault:
+
+  val postgresContainer: TaskLayer[PostgreSQLContainer] =
+    ZLayer.scoped:
+      ZIO.fromAutoCloseable:
+        ZIO.attemptBlocking:
+          val p = PostgreSQLContainer
+            .Def(dockerImageName = DockerImageName.parse("postgres:17.0"))
+            .createContainer()
+          p.start()
+          p
+
+  val dataSource: URLayer[PostgreSQLContainer, DataSource] =
+    ZLayer:
+      ZIO.serviceWith(p =>
+        val ds = PGSimpleDataSource()
+        ds.setUrl(p.jdbcUrl)
+        ds.setUser(p.username)
+        ds.setPassword(p.password)
+        ds
+      )
+
+  val transactor: URLayer[DataSource, Transactor] = Transactor.layer
+
+  val layers: TaskLayer[Transactor] =
+    ZLayer.make[Transactor](
+      postgresContainer,
+      dataSource,
+      transactor
+    )
+
+  override def run: ZIO[ZIOAppArgs with Scope, Any, Any] =
+    ZIO
+      .serviceWithZIO[Transactor] { tx =>
+        tx.transact:
+          sql"create table if not exists car (id serial primary key, model text, top_speed int, vin int, color text, created timestamp)"
+            .update
+            .run()
+
+          sql"create table if not exists owner (id serial primary key, name text, car_id int references car(id))"
+            .update
+            .run()
```

Comment:
It could be nicer if we were using something like `flyway`. Would be more realistic

Reply:
@guizmaii, do you prefer a programmatic way to run `flyway` or maybe a docker-compose way in pairs with the Postges container? I think the second option resonates more with [this](https://github.com/AugustNagro/magnum/issues/90#issuecomment-2608979891) @AugustNagro's comment.

Reply:
Okay. I introduced `docker-compose.yml` with PG and Flyway and moved schema creation to the `magnum-zio-example/sql/migration/V1__create_schema.sql`.

Reply:
> do you prefer a programmatic way to run flyway or maybe a docker-compose way in pairs with the Postges container?

I personally prefer the first one. It allows you to write tests testing your migrations (the docker might allow that to? 🤔)

Reply:
Okay. I changed it to programmatic run the `flyway` migration.